axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java

```diff
@@ -947,6 +957,26 @@ public void cancel(StockMove stockMove, CancelReason cancelReason) throws Axelor
     }
     applyCancelReason(stockMove, cancelReason);
     cancel(stockMove);
+    if (cancellationAutomaticMail == null || cancellationMessageTemplateID == null) {
+      return;
+    }
+    if (isSupplierAutomaticCancellationMail(stockMove, cancellationAutomaticMail)) {
+      sendMailForStockMove(
+          stockMove,
+          Beans.get(TemplateRepository.class)
+              .find(cancellationMessageTemplateID.get().longValue()));
+    }
+  }
+
+  private boolean isSupplierAutomaticCancellationMail(
+      StockMove stockMove, Optional<Boolean> cancellationAutomaticMail) {
+    boolean automaticSending = false;
+    if (cancellationAutomaticMail.isPresent()) {
+      automaticSending = cancellationAutomaticMail.get();
+    }
```

Comment:
To optimise your optional, you should do this : 
boolean automaticSending = cancellationAutomaticMail.getOrElse(false);

I need to check why you have to use an optional here on a boolean.